crates/core/tedge_api/src/entity_store.rs

```diff
@@ -314,12 +347,22 @@ impl EntityStore {
                 }
             }
         };
+
+        let JsonValue::Object(mut other) = message.other else {
+            return Err(Error::EntityRegistrationOtherNotMap);
+        };
+
+        // XXX: this is c8y-specific, entity store shouldn't do this!
+        other
+            .entry("type".to_string())
+            .or_insert(JsonValue::String(self.default_service_type.clone()));
```

Comment:
if the entity store doesn't add this to the payload, then when the mapper is restarted and service.type was changed, then serviceType on all the services would change because the mapper would apply the default again.

Reply:
No no, that's not what I meant. The mapper should be deriving the default `service.type` from the tedge_config value. Not apply a hardcoded value. I prefer to keep it with the mapper, because even for child devices, it's the mapper that applies the default: `thin-edge.io.child` if the registration message doesn't have one specified. Since the mapper is filling one default, it's only better if the mapper fills the other default as well. For this, we need to update the signature of `service_creation_message` to accept `Option<&str>` as `service_type`, just like the `child_device_creation_message` API that accepts `device_type: Option<&str>`.